app-core/src/main/java/net/spals/appbuilder/app/core/generic/GenericWorkerApp.java

```diff
@@ -34,13 +39,21 @@
 
     public static class Builder extends GenericWorkerApp_Builder implements WorkerAppBuilder<GenericWorkerApp> {
 
+        private final List<Module> orderedModules = new ArrayList<Module>();
         private final LifecycleInjectorBuilder lifecycleInjectorBuilder;
         private final ServiceGraph serviceGraph = new ServiceGraph();
```

Comment:
This is actually hand written code. `GenericWorkerApp_Builder` is the FreeBuilder generated code.

It's a little confusing, now that I'm looking at it again. In this case, the FreeBuilder generated code only cares about four pieces of state (all defined in the `App` interface):

1. Logger
2. Name
3. Service config
4. Service Guice injector

The Logger and name are set immediately in the constructor. The constructor also sets a default empty service config. But you can change that via the `setServiceConfig(Config)` or `setServiceConfigFromClasspath(String)` setter methods.

All the member variables here are just working towards creating that last one: service Guice injector. You can see in the `.build()` method the second to last line:
```java
setServiceInjector(buildServiceInjector(lifecycleInjectorBuilder));
```
This is really the guts of the application.